### PR TITLE
#87: Updating/clarifying descriptions for FHAIS scopes and requests. …

### PR DESCRIPTION
**Pull request by MS-ResilienX**
…MitigatorApi: Updating scopes to match FHAIS API scopes.

### CODE REVIEW

**Review by BW-ResilienX — APPROVED** [review]

File: FraihmworkHealthAndIntegrityApi.yaml [hunk 1]
```diff
@@ -352,7 +352,7 @@ paths:
       tags:
       - Components
       summary: Post an update on all component properties
-      description: This information allows FRAIHMWORK to track basic information about a component. Any extra or custom fields are expected to always be present even if null values are provided at first. Using this endpoint means that the component needs to be registered (under the hood), so a UUID will be created and assigned in the response
+      description: This information allows FRAIHMWORK to track basic information about a component. Any extra or custom fields are expected to always be present even if null values are provided at first. Using this endpoint means that the component needs to be registered (under the hood), so a UUID will be created and assigned in the response. Note, requests with the fraihmwork.component.write permission may be limited for authorization reasons. See the descriptions in the securitySchemes.OAuth2.flows.clientCredentials.scopes for more details.
```
Comment (GD-ResilienX) [inline FraihmworkHealthAndIntegrityApi.yaml:355]: Actually, this is the one action where I don't think there are any limitations on the write vs admin. Creating a new component should always work for both of these scopes, barring any bad requests

File: FraihmworkHealthAndIntegrityApi.yaml [hunk 2]
```diff
@@ -678,7 +678,7 @@ paths:
       tags:
       - Faults
       summary: Posts a new fault that is affecting a component
-      description: Allows a component to report one or more faults currently affecting the component. Using this endpoint means that the client is not managing their own fault IDs, and the fault will require a new tracking UUID (that will be created and assigned in the response).
+      description: Allows a component to report one or more faults currently affecting the component. Using this endpoint means that the client is not managing their own fault IDs, and the fault will require a new tracking UUID (that will be created and assigned in the response). Note, requests with the fraihmwork.fault.write permission may be limited for authorization reasons. See the descriptions in the securitySchemes.OAuth2.flows.clientCredentials.scopes for more details.
```
Comment (GD-ResilienX) [inline FraihmworkHealthAndIntegrityApi.yaml:681]: Same as above - posting a fault doesn't have write vs admin differences

File: MitigationApi.yml [hunk 3]
```diff
@@ -58,13 +64,13 @@ paths:
...
-      - MitigationRead:
-        - fraihmwork.mitigation.read
+      - OAuth2: [fraihmwork.mitigation.read]
+      
     post:
       tags:
       - Mitigation
       summary: Posts a mitigation action for one or more faults
-      description: This call adds a system to the collection of possible mitigations. Returns a UUID upon successful POST. Mitigations having MANUAL resolution must only start in ACCEPTED state. Attempts to start mitigations in RECOMMENDED state will return a invalid payload code.
+      description: This call adds a system to the collection of possible mitigations. Returns a UUID upon successful POST. Mitigations having MANUAL resolution must only start in ACCEPTED state. Attempts to start mitigations in RECOMMENDED state will return a invalid payload code. Note, requests with the fraihmwork.mitigation.write permission may be limited for authorization reasons. See the descriptions in the securitySchemes.OAuth2.flows.clientCredentials.scopes for more details.
```
Comment (GD-ResilienX) [inline MitigationApi.yml:73]: Same as above - posting a mitigation doesn't have write vs admin differences

File: MitigationApi.yml [hunk 4]
```diff
@@ -58,13 +64,13 @@ paths:
               schema:
                 $ref: '#/components/schemas/Response'
       security:
-      - MitigationRead:
-        - fraihmwork.mitigation.read
+      - OAuth2: [fraihmwork.mitigation.read]
```
Comment (GD-ResilienX) [inline MitigationApi.yml:67]: read, write and admin

File: MitigationApi.yml [hunk 5]
```diff
@@ -166,14 +184,13 @@ paths:
...
```
Comment (GD-ResilienX) [inline MitigationApi.yml:187]: Read, write, admin

File: MitigationApi.yml [hunk 6]
```diff
@@ -282,8 +310,7 @@ paths:
...
```
Comment (GD-ResilienX) [inline MitigationApi.yml:313]: Read, write, and admin

File: MitigationApi.yml [hunk 7]
```diff
@@ -337,8 +364,7 @@ paths:
...
```
Comment (GD-ResilienX) [inline MitigationApi.yml:367]: Read, write, and admin

File: MitigationApi.yml [hunk 8]
```diff
@@ -387,8 +413,7 @@ paths:
...
```
Comment (GD-ResilienX) [inline MitigationApi.yml:416]: Read, write, and admin

File: MitigationApi.yml [hunk 9]
```diff
@@ -443,8 +468,8 @@ paths:
               schema:
                 $ref: '#/components/schemas/Response'
       security:
-      - MitigationRead:
-        - fraihmwork.mitigation.read       
+      - OAuth2: [fraihmwork.mitigation.read]
```
Comment (GD-ResilienX) [inline MitigationApi.yml:471]: Read, write, and admin

File: MitigationApi.yml [hunk 10]
```diff
@@ -491,8 +516,7 @@ paths:
               schema:
                 $ref: '#/components/schemas/Response'
       security:
-      - MitigationRead:
-        - fraihmwork.mitigation.read
+      - OAuth2: [fraihmwork.mitigation.read]
```
Comment (GD-ResilienX) [inline MitigationApi.yml:519]: Read, write, and admin

File: MitigationApi.yml [hunk 11]
```diff
@@ -547,8 +571,7 @@ paths:
               schema:
                 $ref: '#/components/schemas/Response'
       security:
-      - MitigationManage:
-        - fraihmwork.mitigation.manage
+      - OAuth2: [fraihmwork.mitigation.read]
```
Comment (GD-ResilienX) [inline MitigationApi.yml:574]: Read, write, and admin

File: MitigationApi.yml (hunk 4 above)
Comment (GD-ResilienX) [inline MitigationApi.yml:67]:
```suggestion
      - OAuth2: [fraihmwork.mitigation.read, fraihmwork.mitigation.write, fraihmwork.mitigation.admin]

```

File: MitigationApi.yml (hunk 5 above)
Comment (GD-ResilienX) [inline MitigationApi.yml:187]:
```suggestion
      - OAuth2: [fraihmwork.mitigation.read, fraihmwork.mitigation.write, fraihmwork.mitigation.admin]

```

File: MitigationApi.yml (hunk 6 above)
Comment (GD-ResilienX) [inline MitigationApi.yml:313]:
```suggestion
      - OAuth2: [fraihmwork.mitigation.read, fraihmwork.mitigation.write, fraihmwork.mitigation.admin]

```

File: MitigationApi.yml (hunk 7 above)
Comment (GD-ResilienX) [inline MitigationApi.yml:367]:
```suggestion
      - OAuth2: [fraihmwork.mitigation.read, fraihmwork.mitigation.write, fraihmwork.mitigation.admin]

```

File: MitigationApi.yml (hunk 8 above)
Comment (GD-ResilienX) [inline MitigationApi.yml:416]:
```suggestion
      - OAuth2: [fraihmwork.mitigation.read, fraihmwork.mitigation.write, fraihmwork.mitigation.admin]

```

File: MitigationApi.yml (hunk 9 above)
Comment (GD-ResilienX) [inline MitigationApi.yml:471]:
```suggestion
      - OAuth2: [fraihmwork.mitigation.read, fraihmwork.mitigation.write, fraihmwork.mitigation.admin]

```

File: MitigationApi.yml (hunk 10 above)
Comment (GD-ResilienX) [inline MitigationApi.yml:519]:
```suggestion
      - OAuth2: [fraihmwork.mitigation.read, fraihmwork.mitigation.write, fraihmwork.mitigation.admin]

```

File: MitigationApi.yml (hunk 11 above)
Comment (GD-ResilienX) [inline MitigationApi.yml:574]:
```suggestion
      - OAuth2: [fraihmwork.mitigation.read, fraihmwork.mitigation.write, fraihmwork.mitigation.admin]

```

File: FraihmworkHealthAndIntegrityApi.yaml (hunk 1 above)
Comment (GD-ResilienX) [inline FraihmworkHealthAndIntegrityApi.yaml:355]:
```suggestion
      description: This information allows FRAIHMWORK to track basic information about a component. Any extra or custom fields are expected to always be present even if null values are provided at first. Using this endpoint means that the component needs to be registered (under the hood), so a UUID will be created and assigned in the response. 
```

File: FraihmworkHealthAndIntegrityApi.yaml (hunk 2 above)
Comment (GD-ResilienX) [inline FraihmworkHealthAndIntegrityApi.yaml:681]:
```suggestion
      description: Allows a component to report one or more faults currently affecting the component. Using this endpoint means that the client is not managing their own fault IDs, and the fault will require a new tracking UUID (that will be created and assigned in the response).
```

File: MitigationApi.yml (hunk 3 above)
Comment (GD-ResilienX) [inline MitigationApi.yml:73]:
```suggestion
      description: This call adds a system to the collection of possible mitigations. Returns a UUID upon successful POST. Mitigations having MANUAL resolution must only start in ACCEPTED state. Attempts to start mitigations in RECOMMENDED state will return a invalid payload code.

```

**Review by GD-ResilienX — CHANGES_REQUESTED** [review]

**Review by GD-ResilienX — APPROVED** [review]

**Pull request merged**